### PR TITLE
AI Review: use ΔScore in calculation of top 3 moves

### LINKED ISSUE #1343: Enhancement: Use ΔScore in calculation of top 3 moves for handicap games
I believe score is a better indicator of major events in the game, but this is especially prominent in handicap games.  See discussion at [Top 3 Moves: Score a Better Metric?](https://forums.online-go.com/t/top-3-moves-score-a-better-metric/32702/9).  I assume this would be a backend change, but if I'm wrong, please let me know!

**Proposed Solution**
For handicap games, instead of the top changes in win rate, display the top changes in score.  I think using win rate still has merit in even games (particularly high level ones).

### PR DESCRIPTION
**Pull request by benjaminpjones**
Fixes #1343

Depends on https://github.com/online-go/goban/pull/43

## Proposed Changes

  - Use score instead of win rate for calculating Key Moves
      - note: will fall back on win_rate if score is not provided
  - Show key moves as red dots for site supporter games
  - Cleanup: remove unused variables in AIReview.tsx

## Screenshots

Before:
Screen Shot 2021-11-26 at 4 32 17 PM

After:
Screen Shot 2021-11-26 at 4 32 31 PM

### CODE REVIEW

File: src/views/Game/AIReview.tsx [hunk 1]
```diff
@@ -952,13 +966,35 @@ export class AIReview extends React.Component<AIReviewProperties, AIReviewState>
...
     }
-    public renderWorstMoveList(): JSX.Element {
-        if (!this.props.game.goban?.engine?.move_tree || !this.ai_review) {
+
+    public getWorstMoveList(): AIReviewWorstMoveEntry[] {
+        const move_tree = this.props.game.goban.engine.move_tree;
+        let worst_moves: AIReviewWorstMoveEntry[];
+        let threshhold: number;
+
+        if (this.ai_review?.scores) {
```
Comment (benjaminpjones) [inline src/views/Game/AIReview.tsx:975]: Perhaps more of this logic should live in `Goban::AIReview::computeWorstMoves` (defaulting to score, falling back on winrate, filtering on threshold). @anoek LMK what you think, given that you'd need to make the corresponding backend changes.

**Comment by benjaminpjones** [comment]
The test failures should be resolved when Goban is updated and version bumped.

File: src/views/Game/AIReview.tsx (hunk 1 above)
Comment (anoek) [reply src/views/Game/AIReview.tsx:975]: Yeah that would be great. The backend uses that same code, so if you update it there it'll sync up nicely.

File: src/views/Game/AIReview.tsx [hunk 2]
```diff
@@ -952,13 +966,35 @@ export class AIReview extends React.Component<AIReviewProperties, AIReviewState>
...
+            worst_moves = computeWorstMoves(move_tree, this.ai_review, /*use_score=*/true);
+            threshhold = -5.0;
+        } else {
+            worst_moves = computeWorstMoves(move_tree, this.ai_review);
+            threshhold = -0.2;
+        }
+
+        const filtered_worst_moves = worst_moves.filter(de => de.delta <= -5.0);
```
Comment (anoek) [inline src/views/Game/AIReview.tsx:983]: Instead of `-5.0` should that be `threshold`?

**Comment by anoek** [comment]
This looks great @benjaminpjones , thanks for the update :) As you mentioned, pushing that `computeWorstMoves` down into `goban` would be a good idea, that way the server and client align.

File: src/views/Game/AIReview.tsx (hunk 2 above)
Comment (benjaminpjones) [reply src/views/Game/AIReview.tsx:983]: Oops yeah, thanks for catching!

File: src/views/Game/AIReview.tsx (hunk 1 above)
Comment (benjaminpjones) [reply src/views/Game/AIReview.tsx:975]: Sounds good, I'll make that update later today

**Comment by anoek** [comment]
Thanks for the improvement!

**Comment by philipkd** [comment]
I was looking into implementing the "more key moves" feature as [suggested here](https://forums.online-go.com/t/top-3-moves-score-a-better-metric/32702/18?u=thoseki). Is the change still a work-in-progress? I no longer see the label "+ N moves"  in production on any of my games, even though I'm a Site Supporter.

**Comment by benjaminpjones** [comment]
> I was looking into implementing the "more key moves" feature as [suggested here](https://forums.online-go.com/t/top-3-moves-score-a-better-metric/32702/18?u=thoseki). Is the change still a work-in-progress? I no longer see the label "+ N moves" in production on any of my games, even though I'm a Site Supporter.

@philipkd I'm not working on the +N moves feature, have at it! Let me know if you need some code points (although this PR should give a pretty good idea about where code changes would be)

Also might point you toward @shinuito's PR on an AI Summary table (also in prod but only visible to mods currently): #1569

**Comment by shinuito** [comment]
I didn’t think I messed with the +N part, but I did wrap a div around it and a toggle to align them side by side, so I could’ve unintentionally messed up some css there 😬

**Comment by benjaminpjones** [comment]
@shinuito no you're fine! I just wanted to point @philipkd toward some example code.

I think I may have gotten rid of the +N moves myself but I'm not sure why.

**Comment by philipkd** [comment]
@benjaminpjones great, thanks!

**Comment by philipkd** [comment]
I have a local proof-of-concept that potentially resolves the "more key moves" question, and I'd like to get feedback on the design. Should I create an Issue or a PR? The code is currently a rough draft.

**Comment by anoek** [comment]
PR, you can open it as a draft PR. Screenshots are appreciated as some folks don't have dev environments setup to check it out themselves :)

**Pull request merged**